lib/core/src/main/java/io/atlasmap/core/DefaultAtlasContextFactory.java

```diff
@@ -82,6 +82,14 @@ public static DefaultAtlasContextFactory getInstance() {
         return factory;
     }
 
+    public static DefaultAtlasContextFactory getInstance(CompoundClassLoader cl) {
+        if (factory == null) {
+            factory = new DefaultAtlasContextFactory();
+            factory.init(cl);
```

Comment:
I guess we'd want to do `init()` even if an instance already exists, following `destroy()`? since it's passing a class loader in. Or check if the class loader is same instance as existing?

Reply:
You are right, getting the life cycle right might be tricky. In 74d3ead I tried another option that would work for me. I have also added a test.